lms/djangoapps/ccx/views.py

```diff
@@ -71,43 +71,60 @@ def wrapper(request, course_id):
         and modifying the view's call signature.
         """
         course_key = CourseKey.from_string(course_id)
+        ccx = None
+        if isinstance(course_key, CCXLocator):
+            # is there a security leak here in not checking that this user is
+            # the coach for this ccx?
+            ccx_id = course_key.ccx
+            ccx = CustomCourseForEdX.objects.get(pk=ccx_id)
+            course_key = ccx.course_id
+
         role = CourseCcxCoachRole(course_key)
         if not role.has_user(request.user):
             return HttpResponseForbidden(
                 _('You must be a CCX Coach to access this view.'))
+
         course = get_course_by_id(course_key, depth=None)
-        return view(request, course)
+        return view(request, course, ccx)
     return wrapper
 
 
 @ensure_csrf_cookie
 @cache_control(no_cache=True, no_store=True, must_revalidate=True)
 @coach_dashboard
-def dashboard(request, course):
+def dashboard(request, course, ccx=None):
```

Comment:
Preserving other dashboard code, which uses the `coach_dashboard` decorator, the expected value for 'course' is an actual Course descriptor.  I could update all the other dashboard views to derive the course from the CCXLocator, but that would be repeating a lot of code many times instead of repeating a small bit of code once or twice.